runtime/compiler/control/CompilationThread.cpp

```diff
@@ -6267,9 +6397,61 @@ TR::CompilationInfoPerThreadBase::preCompilationTasks(J9VMThread * vmThread,
             *aotCachedMethod = javaVM->sharedClassConfig->findCompiledMethodEx1(vmThread, J9_ROM_METHOD_FROM_RAM_METHOD(method), &flags);
             if (!(flags & J9SHR_AOT_METHOD_FLAG_INVALIDATED))
                {
+#ifdef COMPRESS_AOT_DATA
+               J9JITDataCacheHeader *dataCacheHeader = (J9JITDataCacheHeader*)*aotCachedMethod;
+               TR_AOTMethodHeader *aotMethodHeader = (TR_AOTMethodHeader*)((ataCacheHeader + 1);
+               if (aotMethodHeader->flags & TR_AOTMethodHeader_CompressedMethodInCache)
+                  {
+                  int sizeOfCompressedDataInCache = dataCacheHeader->size;
+                  int originalDataSize = aotMethodHeader->compileMethodDataSize + aotMethodHeader->compileMethodCodeSize;
+                  void *originalData = trMemory.allocateHeapMemory(originalDataSize);
+                  int aotMethodHeaderSize = sizeof(J9JITDataCacheHeader) + sizeof(TR_AOTMethodHeader);
+                  memcpy(originalData, *aotCachedMethod, aotMethodHeaderSize);
+                  if (inflateBuffer((U_8 *)(*aotCachedMethod)+aotMethodHeaderSize, sizeOfCompressedDataInCache-aotMethodHeaderSize, (U_8* )(originalData)+aotMethodHeaderSize, originalDataSize-aotMethodHeaderSize) == DECOMPRESSION_FAILED)
```

Comment:
Why not use `dataCacheHeader` instead of `*aotcachedMethod` in this line and the one above it?

Reply:
We type cast `*aotCachedMethod` into `J9JITDataCacheHeader`  so that we can get the size of data in shared class cache and we won't have unclear de-reference into buffer. When we inflate we need a buffer that we are going to `inflate`. Reason I do not chose to use `dataCacheHeader` as it might be misleading as that struct has only two fields (8 Bytes). 
https://github.com/eclipse/openj9/blob/6887baea4c98f2820f42071c29de5f7a7ce31642/runtime/oti/j9nonbuilder.h#L560